### DISCUSSION TITLE
is CoroutineCrudRepository compatible with MDC?

### DISCUSSION THREAD

**Discussion by lucasgust** [micronaut-projects/micronaut-data#2521] [Q&A] [answered]
when I retrieve data from a `CoroutineCrudRepository`, the MDC context is lost. how can this issue be resolved?

both examples here demonstrate this behavior, where the MDC context added by the request [filter](https://github.com/lucasgust/txwithr2dbc/blob/main/driver/http-server/src/main/kotlin/example/micronaut/txwithr2dbc/shared/config/RequestMDC.kt) is lost: https://github.com/lucasgust/txwithr2dbc/blob/main/driven/persistence/src/main/kotlin/example/micronaut/txwithr2dbc/employee/adapter/EmployeeDataAccessAdapter.kt#L23

it's relatively simple to reproduce. just execute `docker-compose.yml`, start the application and use the following commands to trigger logs:
- find one: `curl --location 'http://localhost:8080/employees/{UUID}'`
- find all: `curl --location 'http://localhost:8080/employees'`

ps.: the solution mentioned [here](https://github.com/micronaut-projects/micronaut-core/issues/5533#issuecomment-919862196) using `withContext(MDCContext())` from `kotlinx-coroutines-slf4j` does not work.

**Comment by dstepanov** [comment, marked as answer]
Modify your filter in this way https://docs.micronaut.io/latest/guide/#httpFilterContextPropagation

**Reply by lucasgust** [reply]
@dstepanov using this filter blocks my application after calling the repository

**Reply by dstepanov** [reply]
Can you provide a reproducer?

**Reply by lucasgust** [reply]
sorry, I had implemented `MdcPropagationContext` incorrectly and wasn't expecting that the `Map` could be _null_ (automatic Java/Kotlin conversion). it's now working properly. thanks!
